docs/syntax_and_semantics/generics.md

```diff
@@ -91,3 +91,69 @@ end
 class GenericChild(T) < Parent(T)
 end
 ```
+
+## Generics with variable number of arguments
+
+We may define a Generic class with a variable number of arguments using the [splat operator](./operators.md#splats).
+
+Let's see an example where we define a Generic class called `Foo` and then we will use it with different number of type variables:
+
+```crystal-play
+class Foo(*T)
+  getter content
+
+  def initialize(*@content : *T)
+  end
+end
+
+# 2 type variables:
+# (explicitly specifying type variables)
+foo = Foo(Int32, String).new(42, "Life, the Universe, and Everything")
+
+puts typeof(foo) # => Foo(Int32, String)
```

Comment:
Technically, this is not correct because `# =>` indicates to represent the value of the expression in this line. The return value of `puts` would be `nil`. Usually, we don't use `puts` or any other method that actually prints the value in these examples. It's not really necessary because the annotation in the example code is already sufficient to demonstrate this.
If we want the code to actually execute and print the value, it should be `p` instead of `puts`. `p` uses `#inspect` instead of `#to_s`, which should always be used for these kind of inspection purposes and is implicitly assumed by the `# =>` notation. And `p` actually returns the the value of its argument, so the comment annotation is even valid in a strict sense.

Reply:
oh! Great suggestion!! Now I realize that I've been using `# =>` in an incorrect way 🙃 Thx!!

Reply:
So, I'm removing `puts` from every example but the ones in a `crystal-play` block. So that they can run the examples but also have the expected result in the code itself. For these examples I'm using `p` instead of `puts` so that it's technically correct (as you mentioned in the suggestion)